ddtrace/internal/agent.py

```diff
@@ -38,9 +38,28 @@ def get_stats_url():
     return get_env("dogstatsd", "url", default="udp://{}:{}".format(get_hostname(), get_stats_port()))
 
 
+def verify_url(url):
+    # type: (str) -> compat.parse.ParseResult
+    """Verify that a URL can be used to communicate with the Datadog Agent."""
+    parsed = compat.parse.urlparse(url)
+
+    schemes = ("http", "https", "unix")
+    if parsed.scheme not in schemes:
+        raise ValueError(
+            "Unsupported protocol '%s' in Agent URL '%s'. Must be one of: %s" % (parsed.scheme, url, ", ".join(schemes))
+        )
+    elif parsed.scheme in ["http", "https"] and not parsed.hostname:
+        raise ValueError("Invalid hostname in Agent URL '%s'" % url)
+    elif parsed.scheme == "unix" and not parsed.path:
+        raise ValueError("Invalid file path in Agent URL '%s'" % url)
+
+    return parsed
+
+
 def get_connection(url, timeout=DEFAULT_TIMEOUT):
+    # type: (str, float) -> compat.httplib.HTTPConnection
     """Return an HTTP connection to the given URL."""
-    parsed = compat.parse.urlparse(url)
+    parsed = verify_url(url)
```

Comment:
Should this be done instead when we are setting an `agent_url`, e.g. as you've done in the `Tracer` but also in the `AgentWriter`?

Reply:
Yeah I've added it in both places now since there's the possibility (currently) that the url is pulled in by the `AgentWriter` and not the `Tracer`

Reply:
Was thinking this verify step would be instead done when you initialize the `AgentWriter` if no url is passed in by the Tracer, i.e., when `get_trace_url` is called.

Reply:
Oh yes, that might make a bit more sense!